src/python/gudhi/representations/vector_methods.py

```diff
@@ -356,8 +360,8 @@ def fit(self, X, y = None):
 
         if self.predefined_grid is None:
             if self.resolution is None: # Flexible/exact version
-                events = np.unique(np.concatenate([pd.ravel() for pd in X] + [[-np.inf]], axis=0))
-                self.grid_ = np.array(events)
+                print("Empty list or empty diagrams: evaluation grid only contains -infinity")
```

Comment:
Er, unless I am completely confused, this print is executed even if there is a non-empty list of non-empty diagrams :confused:

Reply:
My bad, I put it in the wrong place, thanks for catching this.